### DISCUSSION TITLE
InterReferenceAnalyzer: Why not to use the MemberRef table stream directly to update references?

### DISCUSSION THREAD

**Discussion by A-r-m-i-n-e** [mkaring/ConfuserEx#439] [Q&A] [answered]
Hi,

My question refers to code in Renamer/Analyzers/InterReferenceAnalyzer.cs, in method - Analyze.
I was wondering why MemberRef entries cannot be analyzed in the same manner as TypeRef entries - that is getting the Rows in MemberRef table and updating references accordingly.

Many thanks,
Armine

**Comment by ElektroKill** [comment, marked as answer]
Hi, it doesn’t use the Rows property of the table since that property only includes MemberRefs which were present in the input assembly. The ones added by ConfuserEx or potential plugins are not. The current approach increases the reach of the analyzer to injected members as well.
